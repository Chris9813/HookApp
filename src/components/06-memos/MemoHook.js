import React, { useMemo, useState } from 'react';
import { useCounter } from '../../hooks/useCounter';
import "../02-useEffect/effects.css";
import { procesoPesado } from '../helpers/procesoPesado';


export const MemoHook = () => {

const [show, setshow] = useState(true)
const {counter, increment} = useCounter(5)
const memoProcesoPesado = useMemo(() => procesoPesado(counter), [ counter ])

return (
    <div>
        <h1>MEMO HOOK</h1>
        <h2>counter { counter }</h2>
        <hr/>
        <p>numero de iteraciones {memoProcesoPesado} </p>
        <button
        className ="btn btn-primary ml-3"
        onClick = {increment}
        >+1</button>
        <button
        onClick = {() => {
        setshow(!show)
        }}
        className= "btn btn-outline-primary ml-4"
        >Show/hide {JSON.stringify(show)}</button>
    </div>
)
}
