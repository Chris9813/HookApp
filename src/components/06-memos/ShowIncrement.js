import React from 'react'

export const ShowIncrement = React.memo(({increment}) => {

    console.log("Comming back bitch");
    return (
        <div>
            <button
            className = "btn btn-primary"
            onClick = {() => {
                increment(5);
            }}
            >Incrementar</button>
        </div>
    )
})
