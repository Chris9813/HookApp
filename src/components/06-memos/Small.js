import React from 'react'


export const Small = React.memo(({value}) => {

    console.log("I`M COMMING BACK");
    

    return (
        <div>
            <small>{value}</small>
        </div>
    )
}
)