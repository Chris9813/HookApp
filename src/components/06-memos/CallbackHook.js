import React, { useCallback, useState } from 'react'
import "../02-useEffect/effects.css"
import { ShowIncrement } from './ShowIncrement'

export const CallbackHook = () => {
    const [counter, setcounter] = useState(10)
    
    
    const increment = useCallback( (num) => {
        setcounter(c => c + num)

    }, [ setcounter ])
    
    return (
        <div>
            <h1>use Callback Hook {counter}</h1>
            <hr/>
            <ShowIncrement increment = {increment} />

        </div>
    )
}
