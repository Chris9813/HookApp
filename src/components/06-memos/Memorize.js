import React, { useState } from 'react';
import { useCounter } from '../../hooks/useCounter';
import "../02-useEffect/effects.css";
import { Small } from './Small';

export const Memorize = () => {

    const [show, setshow] = useState(true)

    const {counter, increment} = useCounter(10)
    return (
        <div>
            <h1>counter <Small value = { counter }/></h1>
            <hr/>

            <button
            className ="btn btn-primary ml-3"
            onClick = {increment}
            >+1</button>

            <button
            onClick = {() => {
                setshow(!show)
            }}
            className= "btn btn-outline-primary ml-4"
            >Show/hide {JSON.stringify(show)}</button>
        </div>
    )
}
