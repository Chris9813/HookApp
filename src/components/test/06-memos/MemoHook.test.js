import React from 'react'
import { shallow } from "enzyme"
import { MemoHook } from "../../06-memos/MemoHook"


describe('pruebas con la alicacion MemoHooks', () => {
const wrapper = shallow(<MemoHook />)

test('debe mostrar la info correctamente ', () => {
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find("h2").text().trim() ).toBe("counter 5")
})

test('should ', () => {
    wrapper.find(".btn").simulate("click")
    expect( wrapper.find("h2").text().trim() ).toBe("counter 6")
})

})
