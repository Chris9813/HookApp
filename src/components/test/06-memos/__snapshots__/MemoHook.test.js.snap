// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con la alicacion MemoHooks debe mostrar la info correctamente  1`] = `
<div>
  <h1>
    MEMO HOOK
  </h1>
  <h2>
    counter 
    5
  </h2>
  <hr />
  <p>
    numero de iteraciones 
    5 
     
  </p>
  <button
    className="btn btn-primary ml-3"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="btn btn-outline-primary ml-4"
    onClick={[Function]}
  >
    Show/hide 
    true
  </button>
</div>
`;
